### PR TITLE
Remove v2-prefix from Quickstart guide

### PR DESCRIPTION
**Pull request by tomjaguarpaw**
Closes https://github.com/haskell/cabal/issues/9086

---
Please include the following checklist in your PR:

* [ ] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#other-conventions).

*Seems irrelevant for a documentation PR*

* [ ] Any changes that could be relevant to users [have been recorded in the changelog](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#changelog).

*[Deemed unnecessary](https://github.com/haskell/cabal/pull/9087#issuecomment-1622152060)*

* [ ] The documentation has been updated, if necessary.

*It* is *the documentation*

* [ ] Include [manual QA notes](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#qa-notes) if your PR relates to cabal-install.

*It doesn't*

Bonus points for added automated tests!

*Bonus points forgone*

### CODE REVIEW

**Review by ffaf1 — APPROVED** [review]

**Review by ulysses4ever — APPROVED** [review]
❤️

**Comment by ulysses4ever** [comment]
I wonder if we need a changelog for this change though.

**Comment by ulysses4ever** [comment]
Right. Perhaps it's fine...

**Comment by tomjaguarpaw** [comment]
I appear to have the ability to merge this, which surprises me! I didn't know I had any maintenance rights. Anyway, should I merge it then?

**Comment by ffaf1** [comment]
Wait for the bot. 48h wait period is to allow other contributors to take a look and chime in, if they have something to say.

**Comment by ulysses4ever** [comment]
For the record, PR template (which is not supposed to be removed) had a reference to the contributing page, which has a section about merge process: https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#github-pull-request-conventions

**Comment by tomjaguarpaw** [comment]
[The PR template](https://github.com/haskell/cabal/blob/master/.github/PULL_REQUEST_TEMPLATE.md) didn't seem particularly relevant for documentation-only changes, and only refers to `CONTRIBUTING.md` as "coding conventions", so I didn't look at it. But I take the point that it is supposed to be included nonetheless, so I will do so.

File: doc/nix-local-build.rst [hunk 1]
```diff
@@ -194,10 +194,10 @@ are not currently built on a per-component basis.
 Where are my build products?
 ----------------------------
 
-A major deficiency in the current implementation of v2-build is that
+A major deficiency in the current implementation of build is that
```
Comment (ulysses4ever) [inline doc/nix-local-build.rst:197]:
```suggestion
A major deficiency in the current implementation of ``cabal build`` is that
```
Looks a bit better but it's up to you. Likewise the next change below.

Comment (tomjaguarpaw) [reply doc/nix-local-build.rst]: OK, done.

**Comment by ulysses4ever** [comment]
@mergify rebase

**Comment by ulysses4ever** [comment]
Documentation only -- good candidate for a backport imo

**Comment by ulysses4ever** [comment]
@Mergify backport 3.10

**Pull request merged**